src/app/inventory/store/d2-item-factory.ts

```diff
@@ -587,9 +592,21 @@ export function makeItem(
   // Extract weapon crafting info from the crafted socket but
   // before building stats because the weapon level affects stats.
   createdItem.craftedInfo = buildCraftedInfo(createdItem, defs);
+
+  // Crafting pattern
+  const patternRecordHash = itemNameToCraftingPatternRecordHash(defs)[createdItem.name];
+  if (patternRecordHash) {
+    createdItem.patternUnlockRecord = profileRecords?.records[patternRecordHash];
+  }
+
   // Deepsight Resonance
-  createdItem.deepsightInfo = buildDeepsightInfo(createdItem, item, itemDef);
-  if (createdItem.deepsightInfo?.extractPattern) {
+  createdItem.deepsightInfo = buildDeepsightInfo(createdItem);
+  if (
+    createdItem.deepsightInfo &&
+    !createdItem.deepsightInfo.attunementObjective.complete &&
```

Comment:
Why should the tooltip go away if the objective is complete? The objective being complete means that extraction is available, once it's extracted the objective goes away entirely.